### PR TITLE
osu!taiko performance point adjustments & effective misses

### PR DESCRIPTION
**Pull request by Lawtrohux**
Based on sentiments and feedback from the community and committee members. Changes are:
- GreatHitWindow's (300s) impact on Accuracy PP has been significantly reduced.
- An SR component has been added to Accuracy PP, which replaces most of the scaling seen within GreatHitWindow. As a tradeoff, HR and EZ now each have their respective buff and nerf within difficulty PP.
- The HDFL Bonus has been clamped to ensure that negative accuracy PP isn't given
- Addition of effective misses - this calculated as a ratio, with maps below 1000 combo receiving a multiplier to their misscount. This is able to be viewed via an attribute.

Values are up [here](https://pp.huismetbenen.nl/rankings/players/ltca2), and have been up for public testing for a number of days now.

---

SR/PP sheet: https://docs.google.com/spreadsheets/d/1whakV2tDv4haWs9uETbSOCDQt1x9UP35dvDHWwy7wwU/edit

As of c1da5091191bfac75eb2b5b3294f58addd1b64a5

### CODE REVIEW

**Review by mangomizer — DISMISSED** [review]

**Review by vunyunt — DISMISSED** [review]
Changes are mostly limited to formula tweaks with no structural changes. Community seems happy with these change from what I can see.

**Review by Horiiizons — DISMISSED** [review]

File: osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs [hunk 1]
```diff
@@ -35,7 +37,10 @@ protected override PerformanceAttributes CreatePerformanceAttributes(ScoreInfo s
             countMeh = score.Statistics.GetValueOrDefault(HitResult.Meh);
             countMiss = score.Statistics.GetValueOrDefault(HitResult.Miss);
 
-            double multiplier = 1.12; // This is being adjusted to keep the final pp value scaled around what it used to be when changing things
+            // The effectiveMissCount is calculated by gaining a ratio for totalSuccessfulHits and increasing the miss penalty for shorter object counts lower than 1000.
+            effectiveMissCount = Math.Max(1.0, 1000.0 / totalSuccessfulHits) * countMiss;
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs:41]: Possible div-by-0 (e.g. on drumroll only maps). Needs to be guarded against

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Comment by smoogipoo** [comment]
SR/PP sheet added to description.

File: osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs (hunk 1 above)
Comment (Lawtrohux) [reply osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs]: oops, applied in [3acbcac](https://github.com/ppy/osu/pull/19853/commits/3acbcac4d1a215bb0e17ebb36c3f93309d018cb0)

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**